### ISSUE TITLE
Check if we need to use golint or other tools we have cover all of its use cases.

### ISSUE THREAD

**Issue by dekiel** [kyma-project/test-infra#5746]
**Description**

Check existing before-commit.sh scripts for provided features.

**Reasons**

We need to understand what features must be provided by separate tools.

**Acceptance Criteria**

- [ ] Deliver complete list of features provided by before-commit.sh
- [ ] Mark feature for which we have a prowjob already.

Parent: Golint is deprecated #4943

**Comment by Ressetkk** [comment]
Pat of this issue has been addressed already in #5834

Commands used in before-commit.sh:
* go test
* goimports
* go vet
* go mod download
* go mod verify
* golint

copy-paste from #5834 

go mod verify provides the check for the dependency if the downloaded dependencies match the go.sum entries. Since we use official GOPROXY, which handles such validations we don't really need to take care of downloading the dependency and validate that information. We can easily ignore this step.
We can ignore gofmt goimports and golint steps, as golangci-lint that we recently started using as standalone job takes care of that pretty good.

We have to provide additional job for each component that runs unit tests on the codebase for test-local step. It's not a big problem and splitting that into separate jobs could increase readability.
We also have to prepare jobs for golangci-lint job that runs on contents of the component.

To sum up:
* Add job `go test ./path/to/component-name` for each component
* Add job `golangci-lint run ./path/to/component-name` for each component

**Comment by dekiel** [comment]
@Ressetkk why using official GOPROXY invalidates need to use go mod verify? What you mean as GOPROXY? It's a official URL where goproxy http server is running?
What about go vet? Does golang-lint takes care that too?

**Comment by Ressetkk** [comment]
I've already explained it
> go mod verify provides the check for the dependency if the downloaded dependencies match the go.sum entries. (...) official GOPROXY, which handles such validations (...)
GOPROXY (https://proxy.golang.org) already ensures that the module program imports has matching checksums in go.mod. No need to verify that locally.

golangci-lint contains `govet` linter https://golangci-lint.run/usage/linters/
